pandas/core/frame.py

```diff
@@ -7808,6 +7808,8 @@ def _reduce(
         self, op, name, axis=0, skipna=True, numeric_only=None, filter_type=None, **kwds
     ):
 
+        assert filter_type is None or filter_type == "bool", filter_type
```

Comment:
does it make sense to add type annotation to filter_type as well? and maybe start a docstring.

also the NotImplementedError in _get_data is now unreachable?

also `out_dtype = "bool" if filter_type == "bool" else None` on L7864 is redundant?

Reply:
filter_type is only passed for "any" and "all", id lean towards getting rid of the kwarg altogether

Reply:
> also the NotImplementedError in _get_data is now unreachable?

that has always been the case, but im happy to get rid of that branch altogether